source/E2E.Tests/Services/Workshops/WorkshopCreationTests.cs

```diff
@@ -0,0 +1,67 @@
+using E2E.Tests.Environment;
+using E2E.Tests.Util;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Settlements;
+using TaleWorlds.CampaignSystem.Settlements.Workshops;
+using Xunit.Abstractions;
+
+namespace E2E.Tests.Services.Workshops;
+
+public class WorkshopCreationTests : IDisposable
+{
+    E2ETestEnvironment TestEnvironment { get; }
+    public WorkshopCreationTests(ITestOutputHelper output)
+    {
+        TestEnvironment = new E2ETestEnvironment(output);
+    }
+
+    public void Dispose()
+    {
+        TestEnvironment.Dispose();
+    }
+
+    [Fact]
+    public void ServerCreateWorkshop_SyncAllClients()
+    {
+        // Arrange
+        var server = TestEnvironment.Server;
+
+        // Act
+        string? WorkshopId = null;
+        server.Call(() =>
+        {
+            Settlement settlement = GameObjectCreator.CreateInitializedObject<Settlement>();
+            var Workshop = new Workshop(settlement, "test");
+            Assert.True(server.ObjectManager.TryGetId(Workshop, out string foundWorkshopId));
+            WorkshopId = foundWorkshopId;
```

Comment:
This removes some lines, just looks slightly better

```suggestion
            Assert.True(server.ObjectManager.TryGetId(Workshop, out WorkshopId));
```